### DISCUSSION TITLE
How do I tell OrchardCore's middleware to invoke next?

### DISCUSSION THREAD

**Discussion by inzi** [OrchardCMS/OrchardCore#10596] [Q&A]
If put an App.Use(...) in front of app.UseOrchardCore, my pipeline is invoked.

But if I put afterwards, it is not. 

This tells me it's not doing an await next();

In an Orchard project, I did this. Only "pipeline before" appears in debug output.

            if (env.IsDevelopment())
            {
                app.Use(async (context, next) => {
                    System.Diagnostics.Debug.WriteLine("pipeline before");
                    await next();
                });
            }
            app.UseOrchardCore();
            if (env.IsDevelopment())
            {
                app.Use(async (context, next) => {
                    System.Diagnostics.Debug.WriteLine("pipeline after");
                    await next();
                });
            }
            
Thanks for any advice.

**Comment by ns8482e** [comment]
` app.UseOrchardCore();`  starts modular, tenant aware pipeline

so to run middleware in modular - tenant aware pipeline - 

Option 1 - Using  Module `Startup.cs`

```cs
    public class Startup : StartupBase
    {
         public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
         {
              app.Use(async (context, next) => {
                  System.Diagnostics.Debug.WriteLine("pipeline after");
                  await next();
              });
         }
    }
```

Option 2 - Using `Configure` extension method of Orchard Builder after `AddOrchardCore` or `AddOrchardCms` in configure services in Host `Startup,cs`

```cs
services.AddOrhardCore()
    .Configure( (app,routes,svc) => {
             app.Use(async (context, next) => {
                  System.Diagnostics.Debug.WriteLine("pipeline after");
                  await next();
              });
    });
```
